itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenarios.java

```diff
@@ -589,6 +590,72 @@ public void testIncrementalAdds() throws IOException {
   }
 
   @Test
+  public void testIncrementalLoadWithVariableLengthEventId() throws IOException, TException {
```

Comment:
ok if insert has three events now

alter (99 )
insert (100)
alter ( 101)
truncate(102)

with the old code the order of execution would be 

insert (100)
alter ( 101)
truncate(102)
alter (99 )

this should anyways create a problem i guess in the test, is that not so ?